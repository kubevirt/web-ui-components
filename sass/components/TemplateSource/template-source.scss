.kubevirt-template-source__overlay {
  display: inline-block;
}

.kubevirt-template-source__source {
  word-wrap: break-word;
}
