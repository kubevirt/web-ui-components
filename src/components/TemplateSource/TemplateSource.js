import React from 'react';
import PropTypes from 'prop-types';

import { Url } from '../Details/Url';
import { getTemplateProvisionSource } from '../../utils/templates';
import { PROVISION_SOURCE_URL } from '../../constants';
import { getName, getNamespace } from '../../utils/selectors';
import { prefixedId } from '../../utils/utils';

const Type = ({ type, source, id, isInline }) => (
  <div id={id} title={source} className={isInline ? 'kubevirt-template-source__overlay' : ''}>
    {type}
  </div>
);

Type.propTypes = {
  type: PropTypes.string.isRequired,
  source: PropTypes.string,
  id: PropTypes.string,
  isInline: PropTypes.bool,
};

Type.defaultProps = {
  source: undefined,
  id: undefined,
  isInline: false,
};

const Source = ({ type, source, id }) => {
  if (!source) {
    return null;
  }

  const sourceElem = type === PROVISION_SOURCE_URL ? <Url url={source} short /> : source;

  return (
    <div id={id} className="kubevirt-template-source__source">
      {sourceElem}
    </div>
  );
};

Source.propTypes = {
  type: PropTypes.string.isRequired,
  source: PropTypes.string,
  id: PropTypes.string,
};

Source.defaultProps = {
  source: undefined,
  id: undefined,
};

export const TemplateSource = ({ template, dataVolumes, detailed }) => {
  const provisionSource = getTemplateProvisionSource(template, dataVolumes);

  if (!provisionSource) {
    return '---';
  }

  const { type, source } = provisionSource;
  const id = prefixedId(getNamespace(template), getName(template));
  const typeId = prefixedId(id, 'type');
  const sourceId = prefixedId(id, 'source');

  if (!detailed) {
    return <Type id={typeId} type={type} source={source} isInline />;
  }

  return (
    <React.Fragment>
      <Type id={typeId} type={type} source={source} />
      <Source id={sourceId} type={type} source={source} />
    </React.Fragment>
  );
};

TemplateSource.propTypes = {
  template: PropTypes.object.isRequired,
  dataVolumes: PropTypes.array,
  detailed: PropTypes.bool,
};

TemplateSource.defaultProps = {
  dataVolumes: [],
  detailed: false,
};
