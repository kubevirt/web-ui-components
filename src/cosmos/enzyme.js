import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
