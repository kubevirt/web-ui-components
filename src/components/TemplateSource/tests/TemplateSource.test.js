import React from 'react';
import { shallow } from 'enzyme';

import { TemplateSource } from '../TemplateSource';
import { default as urlFixtures } from '../fixtures/TemplateSource.fixture';
import { containerTemplate, pxeTemplate } from '../../../tests/mocks/user_template';

const testTemplateSource = ({ props }) => <TemplateSource {...props} />;

describe('<TemplateSource />', () => {
  it('renders correctly', () => {
    const tests = [
      ...urlFixtures,
      {
        props: {
          template: pxeTemplate,
        },
      },
      {
        props: {
          template: pxeTemplate,
          detailed: true,
        },
      },
      {
        props: {
          template: containerTemplate,
        },
      },
      {
        props: {
          template: containerTemplate,
          detailed: true,
        },
      },
    ];
    tests.forEach(fixture => {
      const component = shallow(testTemplateSource(fixture));
      expect(component).toMatchSnapshot();
    });
  });
});
