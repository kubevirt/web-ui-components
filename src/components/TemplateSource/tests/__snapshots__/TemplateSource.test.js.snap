// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateSource /> renders correctly 1`] = `
<Type
  id="myproject-url-template-type"
  isInline={true}
  source="http://rock.mountain.all-images.cz/mirrors/there-was-a-toad/on/a/blue/happy/road/0.12.13/isos/x86_64/dromaeosauridae.iso?a=1&b=caaaaaaaaaaddddddddaaaaaaa#here"
  type="URL"
/>
`;

exports[`<TemplateSource /> renders correctly 2`] = `
<Fragment>
  <Type
    id="myproject-url-template-type"
    isInline={false}
    source="http://rock.mountain.all-images.cz/mirrors/there-was-a-toad/on/a/blue/happy/road/0.12.13/isos/x86_64/dromaeosauridae.iso?a=1&b=caaaaaaaaaaddddddddaaaaaaa#here"
    type="URL"
  />
  <Source
    id="myproject-url-template-source"
    source="http://rock.mountain.all-images.cz/mirrors/there-was-a-toad/on/a/blue/happy/road/0.12.13/isos/x86_64/dromaeosauridae.iso?a=1&b=caaaaaaaaaaddddddddaaaaaaa#here"
    type="URL"
  />
</Fragment>
`;

exports[`<TemplateSource /> renders correctly 3`] = `
<Type
  id="myproject-pxe-template-type"
  isInline={true}
  type="PXE"
/>
`;

exports[`<TemplateSource /> renders correctly 4`] = `
<Fragment>
  <Type
    id="myproject-pxe-template-type"
    isInline={false}
    type="PXE"
  />
  <Source
    id="myproject-pxe-template-source"
    type="PXE"
  />
</Fragment>
`;

exports[`<TemplateSource /> renders correctly 5`] = `
<Type
  id="myproject-container-template-type"
  isInline={true}
  source="fooContainer"
  type="Container"
/>
`;

exports[`<TemplateSource /> renders correctly 6`] = `
<Fragment>
  <Type
    id="myproject-container-template-type"
    isInline={false}
    source="fooContainer"
    type="Container"
  />
  <Source
    id="myproject-container-template-source"
    source="fooContainer"
    type="Container"
  />
</Fragment>
`;
